### DISCUSSION TITLE
Use with a React Native Wallet?

### DISCUSSION THREAD

**Discussion by bricestacey** [wevm/wagmi#1676] [Q&A]
Hi, I like the wagmi API. The React hooks are easy to work with and handle my use cases.

However, I am trying to integrate it directly into a React Native wallet. Are there any community connectors that integrate with something like Ethers? Specifically, I have a private key in-memory with ethers.

I have tried adapting the MockConnector with a JsonRpcProvider. `useConnect` hooks and whatnot work for reading the account. I can call `connect` and get the address. But when I try to use the `useContractWrite` hooks, it says there is no provider. Can anyone point me in the right direction?

**Comment by thegitparticle** [comment]
Wagmi's recent versions are tricky with React Native as its a focus area as far as I know. If you have the private key in-memory and not using external wallets to connect, then it's best to use ethers functions for all your needs. For RPC connections, if you using Alchemy as provider, their SDK works pretty well as well.

**Comment by peterferguson** [comment]
Any updates on how you are going about this? @bricestacey 

Naively is seems like the write functions should work if you use a signer instantiated with the private key but I haven't tried it out yet

Interested to know if you tried this before I get into trying it out?
